src/editor/Renderer.cpp

```diff
@@ -424,16 +434,57 @@ void Renderer::renderLoop() {
 		drawEntConnections();
 
 		isLoading = reloading;
+
+		std::set<int> modelidskip;
 		for (int i = 0; i < mapRenderers.size(); i++) {
 			int highlightEnt = -1;
-			if (pickInfo.valid && pickInfo.mapIdx == i && pickMode == PICK_OBJECT) {
+			Bsp* curMap = mapRenderers[i]->map;
+			bool forceRender = false;
+
+			if (pickInfo.valid && g_app->getSelectedMapId() == i && pickMode == PICK_OBJECT) {
 				highlightEnt = pickInfo.entIdx;
 			}
-			mapRenderers[i]->render(highlightEnt, transformTarget == TRANSFORM_VERTEX, clipnodeRenderHull);
+			
+			if (curMap && curMap->ents.size() && !isLoading)
+			{
+				if (curMap->is_model)
+				{
+					for (int n = 0; n < mapRenderers.size(); n++)
+					{
+						if (n == i)
+							continue;
+
+						Bsp* anotherMap = mapRenderers[n]->map;
+						if (anotherMap && anotherMap->ents.size())
+						{
+							for (int s = 0; s < anotherMap->ents.size(); s++)
+							{
+								Entity* tmpEnt = anotherMap->ents[s];
+								if (tmpEnt->hasKey("model"))
+								{
+									if (!modelidskip.count(s))
+									{
+										if (basename(tmpEnt->keyvalues["model"]) == basename(curMap->path))
+										{
+											curMap->ents[0]->setOrAddKeyvalue("origin", tmpEnt->getOrigin().toKeyvalueString());
+											modelidskip.insert(s);
+											break;
+										}
+									}
+								}
+							}
+						}
+					}
+				}
+			}

```

Comment:
This is way too much nesting. Please break some of this out into other functions or something.

Also what is the purpose of this?

Reply:
Yes it can be simplified.

I think this code change origin if model adding to map?